test/unit/server-statefulset.bats

```diff
@@ -1681,6 +1681,25 @@ load _helpers
   [ "${actual}" = "true" ]
 }
 
+@test "server/standalone-StatefulSet: config checksum annotation defaults to off" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      --show-only templates/server-statefulset.yaml \
+      . | tee /dev/stderr |
+      yq '.spec.template.metadata.annotations["vault.hashicorp.com/config-checksum"] == null' | tee /dev/stderr)
+  [ "${actual}" = "true" ]
+}
+
+@test "server/standalone-StatefulSet: config checksum annotation can be enabled" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      --show-only templates/server-statefulset.yaml \
+      --set 'server.configAnnotation=true' \
+      . | tee /dev/stderr |
+      yq '.spec.template.metadata.annotations["vault.hashicorp.com/config-checksum"] == null' | tee /dev/stderr)
+  [ "${actual}" = "false" ]
```

Comment:
Should we also test the value here? Maybe assert something about its length, or expect a specific hash?

Reply:
I did that initially, but realized that the value might be something noisy and annoying to fix every time we change the config file.

Reply:
We are doing those checks as part of the other tests for this kind of feature: https://github.com/hashicorp/vault-helm/blob/04074311f71fd3a9b39f983553b693363f4a4f22/test/unit/prometheus-servicemonitor.bats#L84-L85. 

I think it is worth it to catch any potential regression where we could the expected defaults.